include/fmt/ranges.h

```diff
@@ -95,12 +95,96 @@ template <typename... Ts> struct conditional_helper {};
 template <typename T, typename _ = void> struct is_range_ : std::false_type {};
 
 #if !FMT_MSC_VER || FMT_MSC_VER > 1800
+
+#  define FMT_DECLTYPE_RETURN(val)  \
+    ->decltype(val) { return val; } \
+    static_assert(true, "a")
```

Comment:
the static assert makes it so that a semi colon is required after the macro, which mostly just helps clang-format handle the code better. the "a" is to silence a clang-tidy warning to replace the statement with a [unary static assert](https://clang.llvm.org/extra/clang-tidy/checks/modernize-unary-static-assert.html). mostly a habit of mine but i'm not sure if it's needed here

Reply:
That's what I thought. Looks like clang-format does indeed make a mess without semicolon so let's keep it but replace "a" with an empty string and add a brief comment explaining what's static_assert for.